cloud/terraform/templates.go

```diff
@@ -305,14 +306,25 @@ echo $PRIVATEIP > /tmp/.ip
 	"PUBLICIP=`ip route get 8.8.8.8 | awk '{printf \"%s\", $NF; exit}'`" + `
 
 
-kubeadm init --apiserver-bind-port ${PORT} --token ${TOKEN} --kubernetes-version v${CONTROL_PLANE_VERSION} \
-             --apiserver-advertise-address ${PUBLICIP} --apiserver-cert-extra-sans ${PUBLICIP} ${PRIVATEIP} \
-             --service-cidr ${SERVICE_CIDR}
+KUBEADM_FLAGS=" --apiserver-bind-port ${PORT} --token ${TOKEN} --kubernetes-version v${CONTROL_PLANE_VERSION}"
+KUBEADM_FLAGS+=" --apiserver-advertise-address ${PUBLICIP} --apiserver-cert-extra-sans ${PUBLICIP} ${PRIVATEIP}"
+KUBEADM_FLAGS+=" --service-cidr ${SERVICE_CIDR}"
 
-# install weavenet
-sysctl net.bridge.bridge-nf-call-iptables=1
-export kubever=$(kubectl version --kubeconfig /etc/kubernetes/admin.conf | base64 | tr -d '\n')
-kubectl apply --kubeconfig /etc/kubernetes/admin.conf -f "https://cloud.weave.works/k8s/net?env.CHECKPOINT_DISABLE=1&env.IPALLOC_RANGE=${POD_CIDR}&disable-npc=true&k8s-version=$kubever"
+if [[ "${CNI_PROVIDER}" == "calico" ]]; then
+  KUBEADM_FLAGS+=" --pod-network-cidr=${POD_CIDR}"
+fi
+
+kubeadm init ${KUBEADM_FLAGS}
+
+if [[ "${CNI_PROVIDER}" == "calico" ]]; then
+	# install calico
+	kubectl apply --kubeconfig /etc/kubernetes/admin.conf -f https://raw.githubusercontent.com/cadmuxe/cluster-api/calico/tf-deployer/spec/calico-v3.0.yaml
+else
```

Comment:
Please check explicitly if the set cni is weavenet, and error out if it's anything unsupported. We better not install a CNI provider that the user did not ask for.